src/lib/es6.d.ts

```diff
@@ -127,7 +127,7 @@ interface ObjectConstructor {
       * @param target The target object to copy to.
       * @param source The source object from which to copy properties.
       */
-    assign<T, U>(target: T, source: U): T & U;
+    assign<T, U>(target: T, source: U | null | undefined): T & U;

```

Comment:
>`T & null` and `T & undefined` should not be a problem, because `null` and `undefined` add no new member to the type?

Yes, it appears so, but...

>In fact, I think the type system could (should?) simplify `T & null & undefined` to just `T`.

It doesn't, hence my initial concern that this would be problematic. But it seems to be fine regardless, in that the instances of `T & null` appear to be usable as `T` everywhere. I'm waiting for the TS team to confirm / deny if this is expected behavior or a bug.

>BTW I don't think `Object.assign` is "meant" to explicitely accept `undefined`.

Yes, this is what I thought as well. I'm not familiar with *all* of `assign`'s uses, so if it's not expected that users will want to use it with `null` / `undefined` sources then it would be nice to have a way to specify that `U` can't be `null`. Perhaps `U extends {}` ? I'm not in front of tsc to try at the moment.